### ISSUE TITLE
Tablist roles, states, and properties: aria-label guidance seems unnecessary

### ISSUE THREAD

**Issue by smhigley** [w3c/aria-practices#2319]
In the tablist's [roles, states, and properties](https://w3c.github.io/aria-practices/#wai-aria-roles-states-and-properties-22) list, the third bullet reads as follows:

> If the tab list has a visible label, the element with role tablist has [aria-labelledby](https://w3c.github.io/aria/#aria-labelledby) set to a value that refers to the labelling element. Otherwise, the tablist element has a label provided by [aria-label](https://w3c.github.io/aria/#aria-label).

The sentence after "Otherwise" seems to imply that a tablist must have an accessible name, and should be given one through `aria-label` if there is no visible name. If the intent is to imply that a name is required, it seems like something worth revisiting since many traditional application tablists do not have names. Especially if there is no visible name, it seems like adding a specific screen-reader-only name should be optional.

I think we should consider removing the "Otherwise..." sentence, or altering it to say "Otherwise, the tablist element may have a label provided by `aria-label`."

**Comment by JAWS-test** [comment]
Yes, especially because [tablist](https://w3c.github.io/aria/#tablist) does not have an "Accessible Name Required".

**Comment by mcking65** [comment]
The section titled "Accessible Name Guidance by Role" starts out:

>Certain elements always require a name, others may usually or sometimes require a name, and still others should never be named. The table below lists all ARIA roles and provides the following information for each :
>
>Necessity of Naming: Indicates how necessary it is for authors to add a naming attribute or element to supplement or override the content of an element with the specified role. This column may include one of the following values:
>* Required Only If  Content Insufficient: An element with this role is named by its descendant content. If aria-label  or aria-labelledby  is applied, content contained in the element and its descendants is hidden from assistive technology users unless it is also referenced by aria-labelledby . Avoid hiding descendant content except in the rare circumstances where doing so benefits assistive technology users. 
>* Required: The ARIA specification requires authors to provide a name; a missing name causes accessibility validators to report an error. 
>* Recommended: Providing a name is strongly recommended. 
>* Discretionary: Naming is either optional or, in the circumstances described in the guidance column, is discouraged. 
>* Do Not Name: Naming is strongly discouraged even if it is technically permitted; often assistive technologies do not render a name even if provided.
>* Prohibited: The ARIA specification does not permit the element to be named; If a name is specified, accessibility validators will report an error. 

We developed those 6 levels of guidance because the ARIA specification only has three choices: required, not required, and prohibitted. Not required simply means allowed, and that doesn't really help authors know whether naming is extremely beneficial, potentially beneficial, or potentially problematic. The spec can say what is "supported" or "allowed", but it is not intended to help people understand what to do with the flexibility given to them by the spec.

For tablist, the table includes:
>Recommended
>* Helps screen reader users understand the context and purpose of the tablist. 
>* Use aria-labelledby  if a visible label is present, otherwise use aria-label . 
>* See the Carousel Design Pattern and Tabs Design Pattern.

Given this, would you still like to have some adjustment to the wording? We could change it to, "Otherwise, it is recommended that ". If we did that, it would need to be changed in many patterns.

**Comment by RobGallo** [comment]
I agree with @smhigley . The current text makes it sound like authors __must__ provide either aria-labeledby or aria-label. The text should be change to indicate that if a visible label exists, authors must provide a label. Otherwise, a label is not required.

**Comment by mcking65** [comment]
@RobGallo wrote:

> I agree with @smhigley . The current text makes it sound like authors **must** provide either aria-labeledby or aria-label. The text should be change to indicate that if a visible label exists, authors must provide a label. Otherwise, a label is not required.

Are you requesting that the naming guidance be changed from recommended to something else?

@smhigley, are you requesting the task force not recommend naming tablists?

@RobGallo, as I mentioned above, the APG naming guidance does not include an option for "not required". The options we have are:

* Recommended: Providing a name is strongly recommended. 
* Discretionary: Naming is either optional or, in the circumstances described in the guidance column, is discouraged. 
* Do Not Name: Naming is strongly discouraged even if it is technically permitted; often assistive technologies do not render a name even if provided.

The current guidance for tablist is "recommended". Please see the naming table for details.

**Comment by RobGallo** [comment]
@mcking65
Based on the options you've given, I suppose the only choice is discretionary. However, that isn't really good guidance. The best guidance is that if the author thinks that anyone needs the tablist to have a (visible) name, everyone should have access to the name (aria-labeledby).

As you know, tablists almost never have names, and there is no negative accessibility impact to that. The only time a tablist should have an accessible name is when the page author has given the tablist a visible label. The reason being parity.

**Comment by craigkovatch** [comment]
Bumping this as it seems like a straightforward change with unanimous consensus, anything blocking it as a revision?
